doc/Pacemaker_Explained/en-US/Ch-Constraints.txt

```diff
@@ -88,19 +89,21 @@ indexterm:[Constraints,Location,node]
 |
 |Positive values indicate the resource should run on this
  node. Negative values indicate the resource should not run on this node.
-
-Values of \+/- +INFINITY+ change "should"/"should not" to "must"/"must not".
+ Values of \+/- +INFINITY+ change "should"/"should not" to "must"/"must not".
 indexterm:[score,Location Constraints]
 indexterm:[Constraints,Location,score]
 
 |resource-discovery
 |always
-|Whether Pacemaker should perform resource discovery
-on this node for the specified resource. Limiting resource discovery to
-a subset of nodes the resource is physically capable of running on
-can significantly boost performance when a large set of nodes are present.
-When pacemaker_remote is in use to expand the node count into the hundreds of
-nodes range, this option should be considered.
+|Whether Pacemaker should perform resource discovery (that is, check whether
+ the resource is already running) for this resource on this node. This should
+ normally be left as the default, so that rogue instances of a service can be
+ stopped when they are running where they are not supposed to be. However,
+ there are two situations where disabling resource discovery is a good idea:
+ when a service is not installed on a node, discovery might return an error;
+ and when Pacemaker Remote is used to scale a cluster to hundreds of nodes,
+ limiting resource discovery to allowed nodes can significantly boost
+ performance.
```

Comment:
"< b>It is up to you as an administrator to ensure that the service can /never/ be active on nodes where resource-discovery is disabled.</ b>"